java/client/src/test/java/glide/api/RedisClientTest.java

```diff
@@ -6207,4 +6212,105 @@ public void lcs_with_len_option() {
+        when(commandManager.<byte[]>submitNewCommand(eq(Dump), eq(arguments), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        CompletableFuture<byte[]> response = service.dump(key);
+        byte[] payload = response.get();
+
+        // verify
+        assertEquals(testResponse, response);
+        assertEquals(value, payload);
+    }
+
+    @SneakyThrows
+    @Test
+    public void restore_returns_success() {
+        // setup
+        byte[] key = "testKey".getBytes();
+        long ttl = 0L;
+        byte[] value = "value".getBytes();
+
+        List<byte[]> arguments = List.of(key, Long.toString(ttl).getBytes(), value);
+
+        CompletableFuture<String> testResponse = new CompletableFuture<>();
+        testResponse.complete(OK);
+
+        // match on protobuf request
+        when(commandManager.<String>submitNewCommand(
+                        eq(Restore), argThat(new ByteArrayArgumentMatcher(arguments)), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        CompletableFuture<String> response = service.restore(key, ttl, value);
+
+        // verify
+        assertEquals(testResponse, response);
+        assertEquals(OK, response.get());
+    }
+
+    @SneakyThrows
+    @Test
+    public void restore_with_restoreOptions_returns_success() {
+        // setup
+        byte[] key = "testKey".getBytes();
+        long ttl = 0L;
+        byte[] value = "value".getBytes();
+        Optional<Long> seconds = Optional.of(10L);
+        Optional<Long> frequency = Optional.of(5L);
+
+        List<byte[]> arguments = new ArrayList<>();
+        arguments.add(key);
+        arguments.add(Long.toString(ttl).getBytes());
+        arguments.add(value);
+        arguments.add("REPLACE".getBytes());
+        arguments.add("ABSTTL".getBytes());
+        arguments.add("IDLETIME".getBytes());
+
+        // Add seconds if present
+        seconds.ifPresent(sec -> arguments.add(Long.toString(sec).getBytes()));
+
+        // Add FREQ and frequency if present
+        arguments.add("FREQ".getBytes());
+        frequency.ifPresent(freq -> arguments.add(Long.toString(freq).getBytes()));
+
+        CompletableFuture<String> testResponse = new CompletableFuture<>();
+        testResponse.complete(OK);
+
+        // match on protobuf request
+        when(commandManager.<String>submitNewCommand(
+                        eq(Restore), argThat(new ByteArrayArgumentMatcher(arguments)), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        CompletableFuture<String> response =
+                service.restore(
+                        key,
+                        ttl,
+                        value,
+                        RestoreOptions.builder()
+                                .hasReplace(true)
+                                .hasAbsttl(true)
+                                .seconds(Optional.of(10L))
+                                .frequency(Optional.of(5L))
```

Comment:
Is there a way to have a lombok builder to unwrap optionals?
```suggestion
                                .seconds(10L)
                                .frequency(5L)
```
Maybe it is possible with a modified setter

Reply:
Or you can use `Long` instead of `long`, which is nullable